pkg/endpointmanager/manager.go

```diff
@@ -219,6 +214,63 @@ func Remove(ep *endpoint.Endpoint) {
 			log.WithError(err).WithField("state", state).Warning("Unable to release endpoint ID")
 		}
 	}
+}
+
+// RuleCacheOwner is an interface representing anything which needs to be
+// notified upon endpoint removal from the endpointmanager.
+type RuleCacheOwner interface {
+	// ClearPolicyConsumers removes references to the specified id from the
+	// policy rules managed by RuleCacheOwner.
+	ClearPolicyConsumers(id uint16) *sync.WaitGroup
+}
+
+// WaitEndpointRemoved waits until all operations associated with Remove of
+// the endpoint have been completed.
+func WaitEndpointRemoved(ep *endpoint.Endpoint, owner RuleCacheOwner) {
+	epRemovedChan := Remove(ep, owner)
```

Comment:
No need for channel return?

Reply:
Can you clarify what you mean by a 'channel return'? Not sure what you mean.

Reply:
Why are you returning a channel on that function if in the end you seem to be waiting for the channel to return a result, why not simply return the result?

Reply:
Are you saying I should do this?

```
// WaitEndpointRemoved waits until all operations associated with Remove of
// the endpoint have been completed.
func WaitEndpointRemoved(ep *endpoint.Endpoint, owner RuleCacheOwner) {
	select {
	case <-Remove(ep, owner):
		return
	}
}
```

I'll add a patch with this, as it's simpler.

Reply:
No, this:
```golang
func WaitEndpointRemoved(ep *endpoint.Endpoint, owner RuleCacheOwner) {
<-Remove(ep, owner)
}
```
or even
```golang
func WaitEndpointRemoved(ep *endpoint.Endpoint, owner RuleCacheOwner) {
 Remove(ep, owner)
}
```

because AFAIK `Remove` is never called async anywhere

Reply:
Remove is called in various locations - the only time where it is needed to wait on the removal to be complete is in the unit tests. Removal of the endpoint ID from the pool can be done asynchronously (i.e., during endpoint delete). It leaves existing code not waiting on the result, but allows for the ID to be released in an asynchronous manner and waited upon if desired.